src/cli/commands/update.js

```diff
@@ -0,0 +1,155 @@
+/* @flow */
+
+import type {Reporter} from '../../reporters/index.js';
+import type Config from '../../config.js';
+import inquirer from 'inquirer';
+import repeat from 'repeating';
+import {MessageError} from '../../errors.js';
+import PackageRequest from '../../package-request.js';
+import {Add} from './add.js';
+import {Install} from './install.js';
+import Lockfile from '../../lockfile/wrapper.js';
+
+export const requireLockfile = true;
+
+export function setFlags(commander: Object) {
+  // TODO: support some flags that install command has
+  commander.usage('update');
+}
+
+type Dependency = {
+  name: string,
+  current: string,
+  wanted: string,
+  latest: string,
+  hint: ?string
+};
+
+type InquirerResponses<K, T> = {[key: K]: Array<T>};
+
+// Prompt user with Inquirer
+async function prompt(choices): Promise<Array<Dependency>> {
+  const answers: InquirerResponses<'packages', Dependency> = await inquirer.prompt([{
+    name: 'packages',
+    type: 'checkbox',
+    message: 'Choose which packages to update.',
+    choices,
+    // Couldn't make it work, I guess I'm missing something here
+    // $FlowFixMe: https://github.com/facebook/flow/blob/f41e66e27b227235750792c34f5a80f38bde6320/lib/node.js#L1197
+    pageSize: process.stdout.rows - 2,
+    validate: (answer) => !!answer.length || 'You must choose at least one package.',
+  }]);
+  return answers.packages;
+}
+
+export async function run(
+  config: Config,
+  reporter: Reporter,
+  flags: Object,
+  args: Array<string>,
+): Promise<void> {
+  const lockfile = await Lockfile.fromDirectory(config.cwd);
+  const install = new Install(flags, config, reporter, lockfile);
+  const [deps] = await install.fetchRequestFromCwd();
+
+  const outdatedDeps = (await Promise.all(deps.map(async ({pattern, hint}): Promise<Dependency> => {
+    const locked = lockfile.getLocked(pattern);
+    if (!locked) {
+      throw new MessageError(reporter.lang('lockfileOutdated'));
+    }
+
+    const {name, version: current} = locked;
+    let latest = '';
+    let wanted = '';
+
+    const normalized = PackageRequest.normalizePattern(pattern);
+
+    if (PackageRequest.getExoticResolver(pattern) ||
+        PackageRequest.getExoticResolver(normalized.range)) {
+      latest = wanted = 'exotic';
+    } else {
+      ({latest, wanted} = await config.registries[locked.registry].checkOutdated(config, name, normalized.range));
+    }
+
+    return ({name, current, wanted, latest, hint});
+  })))
+  .filter(({latest, current}) => latest !== current)
```

Comment:
Instead of chaining can we assign these to variables with descriptive names, kind of hard to follow